controllers/openstackdataplane_controller.go

```diff
@@ -163,31 +184,36 @@ func (r *OpenStackDataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 				deployErrors = append(deployErrors, "role.Name: "+role.Name+" error: "+err.Error())
 			}
 			r.Log.Info("Role", "DeployStrategy.Deploy", role.Spec.DeployStrategy.Deploy, "Role.Namespace", instance.Namespace, "Role.Name", role.Name)
-			if !role.Spec.DeployStrategy.Deploy {
-				_, err := controllerutil.CreateOrPatch(ctx, helper.GetClient(), &role, func() error {
-					r.Log.Info("Reconciling Role", "Role.Namespace", instance.Namespace, "Role.Name", role.Name)
-					helper.GetLogger().Info("CreateOrPatch Role.DeployStrategy.Deploy", "Role.Namespace", instance.Namespace, "Role.Name", role.Name)
-					role.Spec.DeployStrategy.Deploy = instance.Spec.DeployStrategy.Deploy
+			// fmt.Printf("role deployidentifier: %s\n", role.Spec.DeployStrategy.DeployIdentifier)
+			// fmt.Printf("instance deployidentifier: %s\n", instance.Spec.DeployStrategy.DeployIdentifier)
+			if role.IsReady() && role.Spec.DeployStrategy.DeployIdentifier == instance.Spec.DeployStrategy.DeployIdentifier {
+				r.Log.Info("Role successfully deployed", "Role.Namespace", instance.Namespace, "Role.Name", role.Name)
+			} else {
+				if !role.Spec.DeployStrategy.Deploy {
```

Comment:
this line needs to check DeployIdentifier on the role. something like:

```
if !role.Spec.DeployStrategy.Deploy && role.Status.DeployIdentifier != instance.Spec.DeployStrategy.DeployIdentifier {
```

we don't want this to keep setting deploy=true on the role after the role has finished, that creates a loop.